src/index.rs

```diff
@@ -984,4 +1003,198 @@ mod tests {
+                    &actual.path2id[key].id
+                );
+            }
+
+            create_file_at(path.clone(), Some(FILE_SIZE_1), Some(FILE_NAME_1));
+
+            let update = actual
+                .update_all()
+                .expect("Should update index correctly");
+
+            for (key, value) in &actual.path2id {
+                println!(
+                    "===😂😂========update_after_entry Resource ID: {}",
+                    &actual.path2id[key].id
+                );
+            }
+
+            println!(
+                "===✔✔==added=:{}   deleted={}",
+                update.added.len().to_string(),
+                update.deleted.len().to_string()
+            );
+
+            assert_eq!(update.deleted.len(), 0);
+            assert_eq!(update.added.len(), 1);
+        })
+    }
+
+    #[test]
+    fn should_correctly_update_all_using_update_one_resource() {
+        run_test_and_clean_up(|path| {
+            create_file_at(path.clone(), Some(FILE_SIZE_2), Some(FILE_NAME_2));
+            let mut actual = ResourceIndex::build(path.clone());
+
+            let mut all_deleted: HashSet<ResourceId> = HashSet::new();
+            let mut all_added: HashMap<CanonicalPathBuf, ResourceId> =
+                HashMap::new();
+
+            println!("==🌹🌹=update_all using update_one unit tests start==");
+            for (key, value) in &actual.path2id {
+                println!(
+                    "===❤❤========update_one_new_entry Resource ID: {}",
+                    value.id
+                );
+            }
+
+            create_file_at(path.clone(), Some(FILE_SIZE_1), Some(FILE_NAME_1));
+
+            for (old_path, old_index_enry) in &actual.path2id {
+                let one_index_update = actual
+                    .clone()
+                    .update_one(&old_path.clone(), old_index_enry.id)
+                    .expect("Should update one resource correctly");
+
+                for (add_path, &add_id) in one_index_update.added.iter() {
+                    all_added.insert(add_path.clone(), add_id.clone());
+                }
+
+                for delete_id in one_index_update.deleted {
+                    all_deleted.insert(delete_id);
+                }
+            }
+
+            let curr_paths: HashMap<CanonicalPathBuf, DirEntry> =
+                discover_paths(path.clone());
+            let curr_entries = scan_entries(curr_paths);
+
+            for (cur_path, cur_entry) in curr_entries {
+                let one_index_update = actual
+                    .update_one(&cur_path.clone(), cur_entry.id)
+                    .expect("Should update one resource correctly");
+
+                for (add_path, &add_id) in one_index_update.added.iter() {
+                    all_added.insert(add_path.clone(), add_id.clone());
+                }
+
+                for delete_id in one_index_update.deleted {
+                    all_deleted.insert(delete_id);
+                }
+            }
```

Comment:
This test case contains no assertion, what is your expectation of the final state?

Also, please don't commit debugging prints (`println!`).